sdk/go/common/workspace/plugins.go

```diff
@@ -361,6 +362,36 @@ func (source *githubSource) newHTTPRequest(url, accept string) (*http.Request, e
 	return req, nil
 }
 
+func (source *githubSource) getHTTPResponse(
+	getHTTPResponse func(*http.Request) (io.ReadCloser, int64, error),
+	req *http.Request) (io.ReadCloser, int64, error) {
+
+	resp, length, err := getHTTPResponse(req)
+	if err == nil {
+		return resp, length, nil
+	}
+
+	// Wrap 403 rate limit errors with a more helpful message.
+	var downErr *downloadError
+	if !errors.As(err, &downErr) || downErr.code != 403 {
+		return nil, -1, err
+	}
+
+	// This is a rate limiting error only if x-ratelimit-remaining is 0.
+	// https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#exceeding-the-rate-limit
+	if downErr.header.Get("x-ratelimit-remaining") != "0" {
+		return nil, -1, err
+	}
+
+	tryAgain := ""
+	if reset, err := strconv.ParseInt(downErr.header.Get("x-ratelimit-reset"), 10, 64); err == nil {
+		delay := time.Until(time.Unix(reset, 0).UTC())
+		tryAgain = fmt.Sprintf(", try again in %s", delay)
+	}
+
+	return nil, -1, fmt.Errorf("GitHub API rate limit exceeded%s: %w", tryAgain, err)
```

Comment:
This is probably fine as-is, but maybe it would be better if we used the logger for the longer, user facing message instead, with the error being a simple "rate limit exceeded" or similar? The reason for that is largely because we might want to make this error longer and maybe multi-line for better readability, and that doesn't match how error strings typically are in Go. Plus if our logger starts colorizing output in the future, this issue will be slightly more obvious to users.

As a starting point that might require some editing:

```go
logging.Errorf("GitHub API rate limit exceeded")
logging.Errorf("Try again in %v", tryAgain)
if req.Header.Get("Authorization") == "" {
  logging.Infof("TIP: You can set GITHUB_TOKEN to make an authenticated request with a higher rate limit.")
}
return nil, -1, fmt.Errorf("rate limit exceeded: %w", err)
```